avr/cores/megacommand/MCL/SeqStepPage.cpp

```diff
@@ -324,14 +337,52 @@ bool SeqStepPage::handleEvent(gui_event_t *event) {
   if (EVENT_CMD(event)) {
     uint8_t key = event->source - 64;
 
-    if (event->mask == EVENT_BUTTON_PRESSED) {
-      switch (key) {
-      case MDX_KEY_YES:
-        if (!note_interface.notes_count_on())
+    uint8_t step = note_interface.get_first_md_note();
+    switch (key) {
+    case 0x10:
+    case 0x11:
+    case 0x12:
+    case 0x13:
+    case 0x14:
+    case 0x15:
+    case 0x16:
+    case 0x17:
+       if (step == 255) {
           return true;
+        }
+      uint8_t param = MD.currentSynthPage * 8 + key - 0x10;
+      if (event->mask == EVENT_BUTTON_RELEASED) {
+       DEBUG_PRINTLN("here");
+        DEBUG_PRINTLN(key);
+        for (uint8_t n = 0; n < NUM_MD_TRACKS; n++) {
+          if (note_interface.notes[n] == 1) {
+            active_track.clear_step_lock(n, param);
+          }
+        }
+      }
+      if (event->mask == EVENT_BUTTON_PRESSED) {
+        if (active_track.get_track_lock(step, param) == 255) { return true; }
+        for (uint8_t n = 0; n < NUM_MD_TRACKS; n++) {
+          if (note_interface.notes[n] == 1) {
+            active_track.set_track_locks(n, param, MD.kit.params[last_md_track][param]);
+            trig_interface.ignoreNextEvent(key);
+          }
+        }
+      }
+      send_locks(step);
+      break;
+    case MDX_KEY_YES:
```

Comment:
@yatli can you look at this switch statement. MDX_KEY_YES doesn't execute.

feels like i've made an obvious mistake ??

Reply:
patch below fixes. do I need to re-read the C/C++ manual ?